packages/salesforcedx-apex-replay-debugger/src/core/logContext.ts

```diff
@@ -99,6 +113,88 @@ export class LogContext {
     );
   }
 
+  public getHeapDumps(): ApexHeapDump[] {
+    return this.apexHeapDumps;
+  }
+
+  public scanLogForHeapDumpLines(): boolean {
+    const heapDumpRegex = RegExp(/\|HEAP_DUMP\|/);
+    for (const line of this.logLines) {
+      if (heapDumpRegex.test(line)) {
+        const splitLine = line.split('|');
+        const heapDump = new ApexHeapDump(
+          splitLine[3] /* heapDumpId */,
+          splitLine[4] /* className */,
+          splitLine[5] /* namespace */,
+          Number(splitLine[6]) /* line */
+        );
+        this.apexHeapDumps.push(heapDump);
+      }
+    }
+    return this.apexHeapDumps.length > 0;
+  }
+
+  public async fetchOverlayResultsForApexHeapDumps(
+    projectPath: string
+  ): Promise<boolean> {
+    let success = true;
+    try {
+      const orgInfo = await new ForceOrgDisplay().getOrgInfo(projectPath);
+      const requestService = new RequestService();
+      requestService.instanceUrl = orgInfo.instanceUrl;
+      requestService.accessToken = orgInfo.accessToken;
+
+      for (const heapDump of this.apexHeapDumps) {
+        this.session.printToDebugConsole(`fetching: ${heapDump.toString()}`);
+        const overlayActionCommand = new ApexExecutionOverlayResultCommand(
+          heapDump.getHeapDumpId()
+        );
+        let errorString;
+        let returnString;
+        await requestService
+          .execute(overlayActionCommand, RestHttpMethodEnum.Get)
+          .then(
+            value => {
+              returnString = value;
+            },
+            reason => {
+              errorString = reason;
+            }
+          );
+        if (returnString) {
+          heapDump.setOverlaySuccessResult(JSON.parse(
+            returnString
+          ) as ApexExecutionOverlayResultCommandSuccess);
+        } else if (errorString) {
+          try {
+            success = false;
+            const error = JSON.parse(
+              errorString
+            ) as ApexExecutionOverlayResultCommandFailure[];
+            const errorMessage = nls.localize(
+              'heap_dump_error',
+              error[0].message,
+              error[0].errorCode,
+              heapDump.toString()
+            );
+            this.session.errorToDebugConsole(errorMessage);
+          } catch (error) {
+            const errorMessage = `${error}. ${errorString}. ${heapDump.toString()}`;
+            this.session.errorToDebugConsole(errorMessage);
+          }
+        }
+      }
+    } catch (error) {
+      success = false;
+      const result = JSON.parse(error) as OrgInfoError;
```

Comment:
Can this be any other error besides from `ForceOrgDisplay`? Even if you don't have a repro, I think it's best to be conservative here because this is called from `launchRequest`

Reply:
@DatGuyJonathan I'm open to suggestions here. What would you like to see?

Reply:
I have a thought, but I was hoping you'd mull over it and propose something. Did you get a chance to do that?

Reply:
It's reporting the error because it was unable to get the OrgInfo, and it adds the error message to the aforementioned string. I don't know what else to do here aside from report the message. Keep in mind that none of the checkpoint processing failing will prevent the replay debugger from starting it's session, it is reported to the user and the debugging sessions will happen but without the heap dumps (or if it were just one of several that it couldn't get, without that particular one).

Reply:
Ok, since you're sure all the errors are caught, there's nothing to change here.